controllers/djob_controller_databricks.go

```diff
@@ -88,13 +92,44 @@ func (r *DjobReconciler) delete(instance *databricksv1alpha1.Djob) error {
 
 	jobID := instance.Status.JobStatus.JobID
 
+	_, err := getJob(r, jobID)
+
 	// Check if the job exists before trying to delete it
-	if _, err := r.APIClient.Jobs().Get(jobID); err != nil {
+	if err != nil {
 		if strings.Contains(err.Error(), "does not exist") {
 			return nil
 		}
 		return err
 	}
 
-	return r.APIClient.Jobs().Delete(jobID)
+	return trackExecutionTime(djobDeleteDuration, func() error {
+		return r.APIClient.Jobs().Delete(jobID)
+	})
+}
+
+func getJob(r *DjobReconciler, jobID int64) (job models.Job, err error) {
+	return trackJob(djobGetDuration, djobGetSuccess, djobGetFailure, func() (models.Job, error) {
+		return r.APIClient.Jobs().Get(jobID)
+	})
+}
+
+func createJob(r *DjobReconciler, instance *databricksv1alpha1.Djob) (job models.Job, err error) {
+	return trackJob(djobCreateDuration, djobCreateSuccess, djobCreateFailure, func() (models.Job, error) {
+		return r.APIClient.Jobs().Create(*instance.Spec)
+	})
+}
+
+func trackJob(duration prometheus.Histogram, success prometheus.Counter, failure prometheus.Counter, 
```

Comment:
nit: I'm 70% on this one. As the `trackJob` func is in the same package as the `vars` are declared for the metrics it should be able to use them directly rather than taking them as params into the method. Might make things cleaner.

Reply:
the thing here was I was trying to avoid having the same code duplicated in both the `createJob` and `getJob` methods. I refactored the logic into `trackJob` to keep it DRY